ch_pipeline/processing/base.py

```diff
@@ -361,6 +361,55 @@ def pending(self):
 
         return pending
 
+    def completed(self):
+        """Find all jobs which have been run and finished
+        with status "FINISHED" or "CRASHED". This is slow.
+
+        Returns
+        -------
+        finished : list
+            Return the tags of all successful jobs.
+        crashed : list
+            Return the tags of all failed jobs.
+        """
+        base = self.base_path
+        working_path = base / "working"
+        crashed_path = base / "crashed"
+
+        if not base.exists():
+            raise ValueError("Base path %s does not exist." % base)
+
+        file_regex = re.compile("^%s$" % self.tag_pattern)
+
+        def _read_status(path: "Path") -> str:
+            file = path / "job" / "STATUS"
+            with file.open() as fh:
+                return fh.readline().strip()
+
+        # Recursively search the target directory
+        # fmt: off
+        entries = [
+            [path.name, _read_status(path)] 
+            for path in working_path.rglob("*") 
```

Comment:
```suggestion
            for path in working_path.glob("*") 
```
One thing that might be slowing is that the `rglob` would be letting it recurse far deeper than it needs to. I don't think you need to go any deeper than the first level.

Within the `crashed` directory you probably want to let it go one level deeper, but you'll again just want a normal glob rather than `rglob` I would guess.